plugin/trino-hive-hadoop2/src/test/java/io/trino/plugin/hive/s3select/TestHiveFileSystemS3SelectPushdownWithSplits.java

```diff
@@ -0,0 +1,149 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.s3select;
+
+import com.google.common.collect.ImmutableList;
+import io.airlift.units.DataSize;
+import io.trino.plugin.hive.HiveConfig;
+import io.trino.spi.connector.ColumnHandle;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.testing.MaterializedResult;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.LongStream;
+
+import static io.airlift.units.DataSize.Unit.KILOBYTE;
+import static io.trino.plugin.hive.HiveColumnHandle.ColumnType.REGULAR;
+import static io.trino.plugin.hive.HiveColumnHandle.createBaseColumn;
+import static io.trino.plugin.hive.HiveFileSystemTestUtils.filterTable;
+import static io.trino.plugin.hive.HiveFileSystemTestUtils.getSplitsCount;
+import static io.trino.plugin.hive.HiveFileSystemTestUtils.newSession;
+import static io.trino.plugin.hive.HiveType.HIVE_INT;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.testing.QueryAssertions.assertEqualsIgnoreOrder;
+import static org.testng.Assert.assertTrue;
+
+public class TestHiveFileSystemS3SelectPushdownWithSplits
+{
+    private String host;
+    private int port;
+    private String databaseName;
+    private String awsAccessKey;
+    private String awsSecretKey;
+    private String writableBucket;
+    private String testDirectory;
+    private SchemaTableName tableWithSplits;
+
+    @Parameters({
+            "hive.hadoop2.metastoreHost",
+            "hive.hadoop2.metastorePort",
+            "hive.hadoop2.databaseName",
+            "hive.hadoop2.s3.awsAccessKey",
+            "hive.hadoop2.s3.awsSecretKey",
+            "hive.hadoop2.s3.writableBucket",
+            "hive.hadoop2.s3.testDirectory",
+    })
+    @BeforeClass
+    public void setup(String host, int port, String databaseName, String awsAccessKey, String awsSecretKey, String writableBucket, String testDirectory)
+    {
+        this.host = host;
+        this.port = port;
+        this.databaseName = databaseName;
+        this.awsAccessKey = awsAccessKey;
+        this.awsSecretKey = awsSecretKey;
+        this.writableBucket = writableBucket;
+        this.testDirectory = testDirectory;
+
+        tableWithSplits = new SchemaTableName(databaseName, "trino_s3select_test_csv_scan_range_pushdown");
+    }
+
+    @DataProvider(name = "testSplitSize")
+    public static Object[][] splitSizeParametersProvider()
+    {
+        return new Object[][] {{3, 2, 15, 30}, {50, 30, 2, 4}};
```

Comment:
`50, 30, 2, 4`

`2` is definitely not going to be the case because we have 3 files . Maybe 

`50, 30, 3, 4`

Reply:
For the scenario 50, 30, both maxSplitSize and maxInitialSplitSize are higher than the actual file size.
This means it will be to 1 split per file, so we will always have 3 splits.
The comparison is done like *lowerBound < splitCount < upperBound*. Given that these are integers, the comparison with (2,4) range is equivalent with splitCount = 3.